### PR TITLE
zcash_primitives: Add Orchard bundle metadata to transaction builder.

### PR DESCRIPTION
**Pull request by nuttycom**
This is best reviewed with whitespace changes hidden.

### CODE REVIEW

File: zcash_primitives/src/transaction/builder.rs [hunk 1]
```diff
@@ -239,6 +239,30 @@ impl BuildConfig {
...
+/// shielded bundles.
+#[derive(Debug)]
+pub struct BuildResult {
+    transaction: Transaction,
+    sapling_meta: SaplingMetadata,
+    orchard_meta: orchard::builder::BundleMetadata,
+}
+
+impl BuildResult {
+    pub fn transaction(&self) -> &Transaction {
```
Comment (daira) [inline zcash_primitives/src/transaction/builder.rs:253]: Add documentation.

**Review by daira — APPROVED** [review]
utACK

**Review by str4d — APPROVED** [review]
utACK cc39bf4429359acfba7c2fbbb0ed202c482420b0

**Review by daira — COMMENTED** [review]
Post-hoc ACK

**Pull request merged**